Source/Dafny/AST/Printer.cs

```diff
@@ -2843,6 +2842,8 @@ public class Printer {
       switch (pat) {
         case IdPattern idPat:
           if (idPat.Id.StartsWith(BuiltIns.TupleTypeCtorNamePrefix)) {
+          } else if (idPat.Id.StartsWith("_")) {
```

Comment:
As I understand,  variables can't start with  the underscore: compiler complains on that.  Also the Reference says: _ident  tokens that begin with an _ are not permitted as user identifiers._ [2.5.2]

But in  this code we  refer to  printing case statement  patterns, not variable  names. There  is a  special  case pattern  `_`.  When  Dafny parses  the   program,  it  automatically  assigns   a  unique  string identifier  for such  patterns. This  identifier has  the form  `_v1`, `_v2`, etc. When program gets  Rprinted, those identifiers are printed instead of original `_` symbol. This breaks the further compilation.

By introducing this patch, I  replace those `_v` identifiers with good old `_`.

Reply:
Can you write a comment (like a 1-sentence summary) in the code about that, so that no more question would be asked about this down the road?

Reply:
Done.